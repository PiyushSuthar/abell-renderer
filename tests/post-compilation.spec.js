const { prefixHtmlTags } = require('../src/post-compilation');

describe('prefixHTMLTags()', () => {
  it('should prefix a single html tag', () => {
    const testHTML = `
    <div class="something" id="something" aria-label=testitem>My Item

    </div>
    `;

    expect(prefixHtmlTags(testHTML, 'testhash')).toMatchSnapshot();
  });

  it('should prefix an auto closing tag', () => {
    const testHtml = `<br class="something" />`;
    expect(prefixHtmlTags(testHtml, 'testhash')).toMatchSnapshot();
  });

  it('should not prefix a closing tag', () => {
    const testHtml = `</div>`;
    expect(prefixHtmlTags(testHtml, 'testhash')).toEqual(testHtml);
  });

  it('should not prefix Doctype', () => {
    const testHtml = `<!DOCTYPE html>;`;
    expect(testHtml).toEqual(testHtml);
  });

  it('should prefix deeply nested tree', () => {
    const testHtml = `
      <div>
        <ul>
          <li><a href="test">test</a><a href="test">test</a></li>
          <li><a href="test">test</a><a href="test">test</a></li>
          <li><a href="test">test</a><a href="test">test</a></li>
          <li><a href="test">test</a><a href="test">test</a></li>
          <li><a href="test">test</a><a href="test">test</a></li>
        </ul>
        <div>
          <h1>test</h1>
          <div>
            <ol>
              <ul>
                <li>testing</li>
                <li>testing</li>
                <li>testing</li>
              </ul>
            </ol>
          </div>
        </div>
      </div>
    `;
    expect(prefixHtmlTags(testHtml, 'testhash')).toMatchSnapshot();
  });

  it('should prefix html with attributes having numbers in it', () => {
    const testHtml = `<div tabindex=0>test</div>`;
    expect(prefixHtmlTags(testHtml, 'testhash')).toMatchSnapshot();
  });
});
