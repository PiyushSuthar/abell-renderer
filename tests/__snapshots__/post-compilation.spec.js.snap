// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefixHTMLTags() should prefix a single html tag 1`] = `
"
    <div class=\\"something\\" id=\\"something\\" aria-label=testitem data-abell-testhash>My Item

    </div>
    "
`;

exports[`prefixHTMLTags() should prefix an auto closing tag 1`] = `"<br class=\\"something\\" data-abell-testhash />"`;

exports[`prefixHTMLTags() should prefix deeply nested tree 1`] = `
"
      <div data-abell-testhash>
        <ul data-abell-testhash>
          <li data-abell-testhash><a href=\\"test\\" data-abell-testhash>test</a><a href=\\"test\\" data-abell-testhash>test</a></li>
          <li data-abell-testhash><a href=\\"test\\" data-abell-testhash>test</a><a href=\\"test\\" data-abell-testhash>test</a></li>
          <li data-abell-testhash><a href=\\"test\\" data-abell-testhash>test</a><a href=\\"test\\" data-abell-testhash>test</a></li>
          <li data-abell-testhash><a href=\\"test\\" data-abell-testhash>test</a><a href=\\"test\\" data-abell-testhash>test</a></li>
          <li data-abell-testhash><a href=\\"test\\" data-abell-testhash>test</a><a href=\\"test\\" data-abell-testhash>test</a></li>
        </ul>
        <div data-abell-testhash>
          <h1 data-abell-testhash>test</h1>
          <div data-abell-testhash>
            <ol data-abell-testhash>
              <ul data-abell-testhash>
                <li data-abell-testhash>testing</li>
                <li data-abell-testhash>testing</li>
                <li data-abell-testhash>testing</li>
              </ul>
            </ol>
          </div>
        </div>
      </div>
    "
`;

exports[`prefixHTMLTags() should prefix html with attributes having numbers in it 1`] = `"<div tabindex=0 data-abell-testhash>test</div>"`;
